src/Randomizer.SMZ3.Tracking/Configuration/LocationInfo.cs

```diff
@@ -124,5 +125,23 @@ public bool IsAccessible(World world, Progression progression)
         /// </summary>
         /// <returns>A string representation of this location.</returns>
         public override string? ToString() => Name[0];
+
+        /// <summary>
+        /// Denotes the block of memory that this location is associated
+        /// with
+        /// </summary>
+        public LocationMemoryType? MemoryType { get; set; }
+
+        /// <summary>
+        /// The memory address offset stored as a string hex value
+        /// for where the location is stored in memory
+        /// </summary>
+        public string? MemoryAddress { get; set; }
```

Comment:
My biggest concern is typing **MemoryAddress** and **MemoryFlag** as strings rather than ints. 

I know you can't use hexadecimal literals in JSON because JSON kinda sucks. I've been considering updating everything to YAML (#94) but you could already make that change if that makes this easier.

YAML is a superset of JSON so it should work as-is without changing the contents, but we'd have to test that everything is still read correctly.

Alternatively we could use some custom JSON converters to parse hex strings into ints in here, but I prefer just using YAML libraries for now. We can convert the configs into proper YAML later.

Reply:
I'll do some additional testing, but a quick 15 minute test makes it look like getting YamlDotNet to be able to deserialize our json may be a bit tricky. It's having issues converting the collections in Tracker.json, even after making them writable with public setters. I could test with maybe converting the current json to yaml, though.

I suppose I could go through and convert everything to ints if preferred, though. I originally did it to be consistent with the emo tracker, but it actually wasn't super consistent in the end anyway. It used ints in some places and hex in others.

Reply:
Oh yeah, I did have some custom converters that it's probably not picking up. I prefer having these as ints but it's not really a big deal if it's not that easy to do now.

Reply:
I've updated everything to ints for now.

At this point the core development is done, and I've tested. I'll probably create a beta and upload to my google drive for folks to test before creating a genuine release. In the meantime while folks test, I'd like to try to have tracker comment on a couple things:

- Automatically detect deaths (and maybe special death messages if against particular bosses)
- When changing between games

In the end I'd like to maybe have tracker comment on special things like performing certain tricks, falling down from Moldorm, etc. I think many should be doable to some degree.